### ISSUE TITLE
Error in MPI_Sendrecv_replace

### ISSUE THREAD

**Issue by SpinTensor** [open-mpi/ompi#8909]
Thank you for taking the time to submit an issue!

### What version of Open MPI are you using?
Git branch master. Commit hash=9f74df98a1e7d2f774c1303457fdea36d44a7917

### Describe how Open MPI was installed:
Cloned the Git-repo and installed with
`./autogen.pl`
`../ompi_src/configure --prefix=/home/fuhl/software/ompi/ --enable-mpi-fortran --disable-man-pages`

### If you are building/installing from a git clone, please copy-n-paste the output from `git submodule status`.
```
git submodule status
 e46276629146e13c2f08ee9f71ba8133b4a0a7d7 3rd-party/openpmix (v1.1.3-2960-ge4627662)
 b6ddc6e36dc88d1599f130886c915909327b0b8d 3rd-party/prrte (dev-31155-gb6ddc6e36d)
```

### Please describe the system on which you are running

* Operating system/version: Ubuntu 20.04.2 LTS
* Computer hardware: One x86 desktop computer 
* Network type: -
* Compiler: gcc (Ubuntu 9.3.0-17ubuntu1~20.04) 9.3.0

-----------------------------

## Details of the problem
Please consider the following short program that utilizes MPI_Sendrecv_replace:
```
#include <mpi.h>
#include <stdlib.h>

int main(int argc, char** argv) {
   MPI_Init(&argc, &argv);
   // Get number or processes
   int comm_size;
   MPI_Comm_size(MPI_COMM_WORLD, &comm_size);
   // Get rank of process
   int my_rank;
   MPI_Comm_rank(MPI_COMM_WORLD, &my_rank);
   // allocating send/recv buffer
   int nints = 4;
   int* srbuffer = (int*) malloc(nints*sizeof(int));
   for (int i=0; i<nints; i++) {srbuffer[i]=my_rank;}
   // Messaging
   MPI_Status mystat;
   int destrank = (my_rank + 1) % comm_size;
   int sourcerank = my_rank - 1;
   if (sourcerank < 0) {
      sourcerank = comm_size - 1;
   }   
   MPI_Sendrecv_replace(srbuffer, nints, MPI_INT,
                        destrank, 0,
                        sourcerank, 0,
                        MPI_COMM_WORLD, &mystat);
   free(srbuffer);
   srbuffer=NULL;
   MPI_Finalize();
   return EXIT_SUCCESS;
}
```
Compile with: `mpicc -o sendrecv_replace sendrecv_replace.c`
Execute with: `mpirun -np 2 ./sendrecv_replace`

The Program will abort and output:
```
[hephaistos:00000] *** An error occurred in MPI_Sendrecv_replace
[hephaistos:00000] *** reported by process [2219245569,1]
[hephaistos:00000] *** on communicator MPI_COMM_WORLD
[hephaistos:00000] *** MPI_ERR_BUFFER: invalid buffer pointer
[hephaistos:00000] *** MPI_ERRORS_ARE_FATAL (processes in this communicator will now abort,
[hephaistos:00000] ***    and MPI will try to terminate your MPI job as well)
[hephaistos:00000] *** An error occurred in MPI_Sendrecv_replace
[hephaistos:00000] *** reported by process [2219245569,0]
[hephaistos:00000] *** on communicator MPI_COMM_WORLD
[hephaistos:00000] *** MPI_ERR_BUFFER: invalid buffer pointer
[hephaistos:00000] *** MPI_ERRORS_ARE_FATAL (processes in this communicator will now abort,
[hephaistos:00000] ***    and MPI will try to terminate your MPI job as well)
[hephaistos:610792] PMIX ERROR: UNREACHABLE in file ../../../../ompi_src/3rd-party/openpmix/src/server/pmix_server.c at line 2962
```

**Comment by ggouaillardet** [comment]
Duplicate of #8907 

Fix is in #8908

**Comment by SpinTensor** [comment]
Sorry, I didn't find it when I was looking for a duplicate. Thanks.
